### PR TITLE
[front] - chore: retrieve light `User`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at enabling to retrieve a dry version of `User`. 

This is done such that we can fetch all the users of a workspace and display them in the `VaultCreationModal`. As of now we cannot perform server-side pagination along with filtering. For this reason we want to fetch all the members of a workspace and perform the pagination/filtering on the client-side.

**References:**
- https://github.com/dust-tt/tasks/issues/1284

## Risk

Non negligible as It touches endpoints

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/vaults/CreateVaultModal.tsx [hunk 1]
```diff
@@ -63,7 +69,10 @@ export function CreateVaultModal({
     disabled: true, // Disable as we just want the mutation function
   });
   const router = useRouter();
-  const { members, isMembersLoading } = useMembers(owner);
+  const { members, isMembersLoading } = useMembers({
+    owner,
+    returnLight: true,
```
Comment (flvndvd) [inline front/components/vaults/CreateVaultModal.tsx:74]: This name feels weird. Can we simply go with either mode: "light" or something more concise?

File: front/lib/swr/memberships.ts [hunk 2]
```diff
@@ -11,18 +11,24 @@ import {
 import type { GetWorkspaceInvitationsResponseBody } from "@app/pages/api/w/[wId]/invitations";
 import type { GetMembersResponseBody } from "@app/pages/api/w/[wId]/members";
 
-export function useMembers(
-  owner: LightWorkspaceType,
-  pagination?: PaginationState
-) {
+export function useMembers<T extends boolean = false>({
```
Comment (flvndvd) [inline front/lib/swr/memberships.ts:14]: Let's use a better name than T here.

File: front/lib/swr/memberships.ts [hunk 3]
```diff
@@ -11,18 +11,24 @@ import {
...
   appendPaginationParams(params, pagination);
 
-  const membersFetcher: Fetcher<GetMembersResponseBody> = fetcher;
-  const { data, error, mutate } = useSWRWithDefaults(
-    `/api/w/${owner.sId}/members`,
-    membersFetcher
-  );
+  const url = returnLight
+    ? `/api/w/${owner.sId}/members?light=true`
+    : `/api/w/${owner.sId}/members`;
```
Comment (flvndvd) [inline front/lib/swr/memberships.ts:28]: Can we use queryParams to build the parameters?

File: front/pages/api/w/[wId]/members/index.ts [hunk 4]
```diff
@@ -31,6 +32,8 @@ async function handler(
         supportedOrderColumn: ["createdAt"],
       });
 
+      const returnLight = req.query.light as string;
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/members/index.ts:35]: No casting please.

```
const {light} = req.query;

if (typeof light !== "string") {
 // handle error.
}
```

**Review by flvndvd — COMMENTED** [review]
I remember edging toward this as well, but looking at the implementation details now, I'm wondering if we should not have a "cheap" search version of this endpoint rather than introducing a new type 🤔

**Comment by JulesBelveze** [comment]
As discuss IRL we will create a `/search` endpoint to be able to properly paginate.

**Pull request closed without merging**